src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -646,7 +665,8 @@ AdaptiveExecutor(CitusScanState *scanState)
 		tuplestore_begin_heap(randomAccess, interTransactions, work_mem);
 
 	TransactionProperties xactProperties =
-		DecideTransactionPropertiesForTaskList(distributedPlan->modLevel, taskList);
+		DecideTransactionPropertiesForTaskList(distributedPlan->modLevel, taskList,
```

Comment:
Since it does not depend on current state, maybe just initialize xactProperties directly?